docs/proposals/perf.md

```diff
@@ -0,0 +1,381 @@
+- 1 pod : 0.10 vCPU & 250MB RAM
+
+With KubeEdge pod deployment we can accomodate 10 pods/1vCPU approximately.
+Base on the above K8S Nodes flavor, the CPU and RAM are 32vCPU and 128GB respectively.
+Per K8S Node we should be able to deploy 320 pods(KubeEdge Edge Nodes)/32vCPU
+and RAM consumption would be around 80GB. If we have 5 K8S Nodes with the similar flavor,
+on a whole we should be able to deploy 1500 pods(KubeEdge Edge Nodes)/5 K8S Nodes.
+
+### Performance Test Framework
+<img src="../images/perf/perf-test-framework.png">
+
+KubeEdge Performance Test Framework will be designed based on the **Gomega** and **Ginkgo**.
+
+The Performance Test Framework mainly relates to Utils Library and different types of tests:
+- E2E Test
+- Latency Test
+- Load Test
+- Scalability Test
+- ...
+
+E2E Test Sample:
+
+```
+It("E2E_Test_1: Create deployment and check the pods are coming up correctly", func() {
+			var deploymentList v1.DeploymentList
+			var podlist metav1.PodList
+			replica := 1
+			//Generate the random string and assign as a UID
+			UID = "deployment-app-" + utils.GetRandomString(5)
+			IsAppDeployed := utils.HandleDeployment(http.MethodPost, ctx.Cfg.ApiServer+DeploymentHandler, UID, ctx.Cfg.AppImageUrl[1], nodeSelector, replica)
+			Expect(IsAppDeployed).Should(BeTrue())
+			err := utils.GetDeployments(&deploymentList, ctx.Cfg.ApiServer+DeploymentHandler)
+			Expect(err).To(BeNil())
+			for _, deployment := range deploymentList.Items {
+				if deployment.Name == UID {
+					label := nodeName
+					podlist, err = utils.GetPods(ctx.Cfg.ApiServer+AppHandler, label)
+					Expect(err).To(BeNil())
+					break
+				}
+			}
+			utils.CheckPodRunningState(ctx.Cfg.ApiServer+AppHandler, podlist)
+		})
+```
+
+By default Performance Test Framework will run all tests when the users run the **perf.sh** script.
+Also the users can also provide the specific tests to run as a command line input to the **perf.sh** script.
+
+Performance Test Framework also has the support of a command line interface with plenty of handy command line arguments
+for running your tests and generating test files. Here is a choice example:
+
+    - Ex:   perf.test -focus="LoadTest" and perf.test -skip="ScalabilityTest"
+
+Performance Test Framework features include:
+
+- A comprehensive test runner.
+- Built-in support for testing asynchronicity.
+- Modular and easy to customize.
+- Logging and Reporting.
+- Scalable to add more features.
+- Built-in support of command line interface.
+- ...
+
+### Performance Test Metrics Tools
+* [Prometheus](https://github.com/prometheus/prometheus)
+* [Grafana](https://github.com/grafana/grafana)
+
+### Performance Test Scenarios
+
+#### 1. Edge Nodes join in K8S Cluster
+<img src="../images/perf/perf-edgenodes-join-cluster.png">
+
+Test Cases:
+* Measure Edge Nodes join in K8S Cluster startup time.
+
+  Different numbers of Edge Nodes need be tested.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200...]`.
```

Comment:
if the numbers are same for all the test cases , it is better to mention only once before describing the tests.